### ISSUE TITLE
Version 1.15 Sending large files fails

### ISSUE THREAD

**Issue by Joackk** [localsend/localsend#1506]
### Describe the bug

The program shuts down automatically when the file gets too big. I tried a compression package of about 5.5 gigabytes and did a flash back, I tried a larger one will also do a flash back.It worked fine when I rolled back the version to 1.14

### To reproduce

This is true for sending files larger than 5GB

### Expected behavior

Normal reception

### Desktop (please complete the following information)

wind10

**Comment by Tienisto** [comment]
Do you know how you selected the file? Via drag and drop, via file picker?

**Comment by Joackk** [comment]
> Do you know how you selected the file? Via drag and drop, via file picker?

With the file selector I'm using, he shows that the file has been properly selected, clicking send works fine, and then after clicking accept on the corresponding receiver, he exits the program automatically and without any error message. So I retested version 1.14 and he works fine!

**Comment by sasislm2** [comment]
The app crashed after a few seconds. iOS 18 (Beta 3) to Windows 11. It works fine in v1.14.

**Comment by CATXYLS** [comment]
Yes,

Regardless of whether it is a single file or multiple files, as long as the total transfer size equals or exceeds 2GB, LocalSend will freeze, except for link sharing. Entering the transfer details page causes it to freeze immediately, while not entering the transfer details page results in a failed transfer without freezing.

Whether using the file picker or drag-and-drop method, the bug persists.

My testing environment is Windows 10-22H2 and Android 13, both running the latest released version 1.15 of LocalSend.

Here is my test list:

1. Sending a 5GB file
   - Result: LocalSend freezes or cannot complete the transfer.

2. Sending a 2.5GB file
   - Result: LocalSend freezes or cannot complete the transfer.

3. Sending a 1.8GB file
   - Result: Transfer proceeds normally.

4. Sending a 2.1GB file
   - Result: LocalSend freezes or cannot complete the transfer.

5. Sending a 1.99GB file
   - Result: Transfer proceeds normally.

6. Sending a 2GB file
   - Result: LocalSend freezes or cannot complete the transfer.

7. Receiving a 2GB file (sent from Windows 1.14)
   - Result: LocalSend freezes and crashes.

8. Sending a file just under 2GB (2147483648 bytes - 1 byte)
   - Result: Entering the transfer details page does not cause a freeze, but an error is reported ([500] Could not save file).

9. Sending a file just under 2GB, but the receiving end is Windows 1.14
   - Result: Transfer proceeds normally.

10. Sending four 512MB files
    - Result: LocalSend freezes or cannot complete the transfer.

11. Sending three 512MB files and one 511MB file
    - Result: Transfer proceeds normally.

12. Sharing a 2GB file via link
    - Result: Transfer proceeds normally.

13. Sharing a 6GB file via link
    - Result: Transfer proceeds normally.

Due to time constraints, the above test list is not exhaustive. Please forgive any omissions or uncovered areas.

In summary, when using LocalSend v1.15 for a single transfer session, if the total file size is greater than or equal to 2GB (precisely to the byte), it will freeze and crash. Even if it does not, the transferred file cannot be saved correctly. The bug exists when both the sender and receiver are using v1.15, but if the sender is v1.14 and the receiver is v1.15, the receiver will freeze. It seems that v1.15 cannot handle transfer sessions of 2GB or more, but using links for the transfer does not cause a freeze. This bug appears to occur only between LocalSend clients.

By the way, after freezing, the LocalSend tray icon remains in the taskbar.

> 您知道您是如何选择文件的吗？通过拖放，通过文件选择器？

**Comment by Tienisto** [comment]
Thank you very much. I believe the size calculation exceeds the integer constraint, I will check this further

**Comment by cecatto** [comment]
Noticed the same pattern sending from iOS (17.5.1) with app 1.15.0 to Windows 11 with app 1.15.0.
Downgrading the Windows app to 1.14.0, I was able to send a few files, but anything bigger than 2gb crashes the iOS app (Windows app doesn't crash with 1.14.0).

Tried "share by link", same result.

**Comment by Tienisto** [comment]
Apparently, the recently added library for the taskbar couldn't handle values greater than 32 bit. I've added a workaround in #1512.

Relevant issue in library:
https://github.com/alexmercerind/windows_taskbar/issues/19

**Comment by Tienisto** [comment]
I have added a hot fix: https://github.com/localsend/localsend/releases/download/v1.15.0/LocalSend-1.15.0-windows-x86-64--fix-big-files.exe

Otherwise, you can use the older version 1.14.0, or wait a few days until 1.15.1 is released.

**Comment by fduraibi** [comment]
I still have a problem with large files (tried with 6GB and 8GB and both fails)

Running version 1.15.2

**Comment by CATXYLS** [comment]
> I still have a problem with large files (tried with 6GB and 8GB and both fails)
> 
> Running version 1.15.2

Hello, this issue has been closed. suggest that you create a new issue

**Comment by CATXYLS** [comment]
> I still have a problem with large files (tried with 6GB and 8GB and both fails)
> 
> Running version 1.15.2

Hello,
I transferred 7 GB of files using LocalSend version 1.15.2 and it went smoothly. How did you trigger this issue? 
You could try uninstalling and reinstalling LocalSend. If the problem persists, I recommend creating a new issue and describing in detail how the problem occurs. This will help the developers reproduce and pinpoint the issue.
